src/main/scala/djinni/ObjcppGenerator.scala

```diff
@@ -378,7 +378,8 @@ class ObjcppGenerator(spec: Spec) extends BaseObjcGenerator(spec) {
  }
 
   def writeObjcFile(fileName: String, origin: String, refs: Iterable[String], f: IndentWriter => Unit) {
-    createFile(spec.objcppOutFolder.get, fileName, (w: IndentWriter) => {
+    val folder = if (fileName.endsWith(spec.objcHeaderExt)) spec.objcppHeaderOutFolder else spec.objcppOutFolder
```

Comment:
Choosing the output folder depending on the file extension seems a bit fragile to me. I feel like it would be better to explicitly tell `writeObjcFile` if it is currently writing a header or source file.

But this seems to be how it's done in other places too, so I decided to go the easy way that keeps this PR small. Maybe maybe this should eventually be refactored somewhere in the future, what you think?... 🙈

Reply:
it might be possible to catch that more early, 
https://github.com/cross-language-cpp/djinni-generator/blob/b01a3069e0927f5e0fd681ec13dd20c0150344ea/src/main/scala/djinni/ObjcppGenerator.scala#L338 for the header

for the implementation, https://github.com/cross-language-cpp/djinni-generator/blob/b01a3069e0927f5e0fd681ec13dd20c0150344ea/src/main/scala/djinni/ObjcppGenerator.scala#L357

at least, from reading it looks like that

if the whole file name would be generated there, and `writeObjcFile` would only write to the filename it gets, without applying it's own magic, what should be the case anyway, than that could work

would you like to give that a trial?

Reply:
I decided to introduce a parameter `isHeader`, that tells `writeObjCFile` if what it writes is supposed to go into the header or the source dir. It was pretty straightforward, I may introduce the same concept in other places as well in a separate PR.

Reply:
Not sure about `isHeader`, can you please explain a little bit more

To me it seems, when I can pass `isHeader` and a file name, I could also directly construct the file path.
That would eliminate a parameter, a if branch, and let writeFile make what it should , writing a file, not guessing a directory.
Or is that not possible?

Reply:
The problem is that the filename is constructed differently in every call of `writeObjCFile`:

L56: `objcppMarshal.privateHeaderName(ident.name)`
L96: `objcppMarshal.privateHeaderName(ident.name)`
L145: `privateBodyName(ident.name)`
L338: `objcppMarshal.privateHeaderName(objcName)`
L357: `privateBodyName(objcName)`

This is how `objcName` is defined: `val objcName = ident.name + (if (r.ext.objc) "_base" else "")`

I didn't find a pattern in this that could easily be moved into the write function, that's why the name is still passed in by the caller.

Reply:
I think we can assume that the places where `privateBodyName` is used are the implementation files, `mm`, the other the header.
But we put this as an extra topic for the future, we could also review the same in the objectivec part, where it is exactly the same.

Thanks for the help understanding the code base better!

I think it is possible to simplify that in some future, but please proceed as you do, or as you have it, since we should not mix too much into that ticket that solves a very specific problem, and that it does well. (and pls feel free to resolve that conversation after reading this)

Reply:
ohh, now I get it... 🤦  `objcppMarshal.privateHeaderName()` and `privateBodyName()` could indeed be moved into `writeObjCFile`. I can then just pass in `ident.name` or `objcName` according to the context. 🙈 
Thx, I'll do that. 

> we could also review the same in the objectivec part

Yep, it's the same there so it should be refactored, but not in this PR 👍